x-pack/test/functional/page_objects/infra_logs_page.ts

```diff
@@ -18,8 +30,26 @@ export function InfraLogsPageProvider({ getPageObjects, getService }: FtrProvide
       await pageObjects.common.navigateToApp('infraLogs');
     },
 
-    async navigateToTab(logsUiTab: LogsUiTab) {
-      await pageObjects.common.navigateToUrlWithBrowserHistory('infraLogs', `/${logsUiTab}`);
+    async navigateToTab<T extends LogsUiTab>(logsUiTab: T, params?: TabsParams[T]) {
```

Comment:
(assuming the types suggested above)

```suggestion
    async navigateToTab(logsUiTab: 'stream', params: LogStreamTabParameters)
    async navigateToTab(logsUiTab: 'log-rate', params: LogRateTabParameters)
    async navigateToTab(logsUiTab: 'log-categories', params: LogCategoriesTabParameters)
    async navigateToTab(logsUiTab: 'settings')
    async navigateToTab(logsUiTab: LogsUiTab, params?: object) {
```

Reply:
It could work. Let me give it a try

Reply:
I tried, and unfortunately the compiler doesn't like the `params?: object` bit when the code tries to access the inside.

<img width="838" alt="Screenshot 2020-03-10 at 20 46 15" src="https://user-images.githubusercontent.com/57448/76352978-3fd87b80-6310-11ea-91e0-1ad1375cee93.png">

It think it needs to have some sort of union of all the possible values in the function declaration.

I think the current way is good enough. Maybe we can revisit this in the future if it bothers people.